### PR TITLE
Deprecate loadsApplicationClasses

### PR DESCRIPTION
**Pull request by stuartwdouglas**
After the ClassLoader change it is no longer required.

### CODE REVIEW

File: core/builder/src/main/java/io/quarkus/builder/BuildContext.java [hunk 1]
```diff
@@ -24,12 +24,14 @@
...
 public final class BuildContext {
+    private final ClassLoader classLoader;
     private final StepInfo stepInfo;
     private final Execution execution;
     private final AtomicInteger dependencies;
     private volatile boolean running;
 
-    BuildContext(final StepInfo stepInfo, final Execution execution) {
+    BuildContext(ClassLoader classLoader, final StepInfo stepInfo, final Execution execution) {
+        this.classLoader = classLoader;
```
Comment (gastaldi) [inline core/builder/src/main/java/io/quarkus/builder/BuildContext.java:34]: Better add an assertion here to make sure this is never null, or weird errors might appear

**Review by gastaldi — APPROVED** [review]
LGTM, let's see what @dmlloyd says

**Comment by gsmet** [comment]
@dmlloyd any chance you could have a look to that one?

**Review by dmlloyd — APPROVED** [review]
Seems OK to me.

**Pull request merged**